### PR TITLE
fix: display correct error message if pool not found

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Displays a specific error message "Pool not available" if our system is not able to find the selected pool. This fix was raised originally in #6382 but is no longer relevant to that PR. At this point this fix is purely theoretical and not likely to manifest in production, but paranoia.

## Pull Request Type

(marked as but, but this fix is theoretical) 

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Addresses https://github.com/shapeshift/web/pull/6382/files#r1529279931

Very low risk.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Engineers test copy and logic with monkey patch.
Ops check that LP add liquidity is still working as expected.

## Screenshots (if applicable)

When happy
image

When sad
image

### CODE REVIEW

File: src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx [hunk 1]
```diff
@@ -455,6 +455,14 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
     refetchInterval: 60_000,
   })
 
+  const poolMissing = useMemo(() => {
+    if (poolAsset === undefined || pools === undefined) {
+      return false
```
Comment (0xApotheosis) [inline src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx:460]: Should this return `true` here for paranoia if we don't have a `poolAsset` or `pools`?

Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx:460]: Seconding this, but only if it is preceeded by loading checks that return false when loading pool data

**Review by gomesalexandre — COMMENTED** [review]
Tested with a monkey patch:

- Pool missing

Screenshot 2024-03-20 at 08 54 21

- No regression with the monkey patch removed

image

Still unsure we want to get this in however following our discussions re: letting users select pools that they can't use is a regression

**Comment by woodenfurniture** [comment]
Closing as it's not actually possible to replicate this with the current asset search

**Pull request closed without merging**